OpenRA.Game/Network/Connection.cs

```diff
@@ -131,57 +136,65 @@ public void Dispose()
 
 	sealed class NetworkConnection : EchoConnection
 	{
-		TcpClient socket;
-		int clientId;
-		ConnectionState connectionState = ConnectionState.Connecting;
-		Thread t;
+		readonly TcpClient tcp;
+		readonly List<byte[]> queuedSyncPackets = new List<byte[]>();
+		volatile ConnectionState connectionState = ConnectionState.Connecting;
+		volatile int clientId = -1;
```

Comment:
Looks like the default used to be 0.  Are you sure this change won't regress &lt;obscure edge case here&gt;?